core/frontend/src/components/parameter-editor/ParameterEditor.vue

```diff
@@ -0,0 +1,312 @@
+          <v-spacer />
+          <v-btn
+            color="blue darken-1"
+            text
+            @click="edit_dialog = false"
+          >
+            Cancel
+          </v-btn>
+          <v-btn
+            color="blue darken-1"
+            text
+            @click="saveParam(edited_param.rebootRequired)"
+          >
+            {{ edited_param?.rebootRequired ? "Save and Reboot" : "Save" }}
+          </v-btn>
+        </v-card-actions>
+      </v-card>
+    </v-dialog>
+  </v-card>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+
+import mavlink2rest from '@/libs/MAVLink2Rest'
+import Notifier from '@/libs/notifier'
+import autopilot_data from '@/store/autopilot'
+import autopilot from '@/store/autopilot_manager'
+import Parameter, { ParamType } from '@/types/autopilot/parameter'
+import { parameters_service } from '@/types/frontend_services'
+import { VForm } from '@/types/vuetify'
+import back_axios from '@/utils/api'
+
+import SpinningLogo from '../common/SpinningLogo.vue'
+
+const notifier = new Notifier(parameters_service)
+
+export default Vue.extend({
+  name: 'ParameterEditor',
+  components: {
+    SpinningLogo,
+  },
+  data() {
+    return {
+      search: '',
+      edited_param: undefined as (undefined | Parameter),
+      edit_dialog: false,
+      edited_bitmask: [] as number[],
+      forcing_input: false,
+      custom_input: false,
+      new_value: 0,
+    }
+  },
+  computed: {
+    params() {
+      return autopilot_data.parameters
+    },
+    finished_loading() {
+      return autopilot_data.finished_loading
+    },
+    form(): VForm {
+      return this.$refs.form as VForm
+    },
+    show_advanced_checkbox(): boolean {
+      return typeof this.isInRange(this.edited_param?.value ?? 0) === 'string'
+    },
+    show_custom_checkbox(): boolean {
+      return !!(this.edited_param?.options || this.edited_param?.bitmask)
+    },
+    as_select_tems() {
+      const entries = Object.entries(this.edited_param?.options ?? [])
+      return entries.map(([value, name]) => ({ text: name, value: parseFloat(value), disabled: false }))
+    },
+  },
+  methods: {
+    isInRange(input: number): boolean | string {
+      if (!this.edited_param?.range) {
+        return true
+      }
+      if (input > this.edited_param.range.high) {
+        return `Value should be smaller than ${this.edited_param.range.high}`
+      }
+      if (input < this.edited_param.range.low) {
+        return `Value should be greater than ${this.edited_param.range.low}`
+      }
+      return true
+    },
+    isValidType(input: number): boolean | string {
+      if (this.edited_param?.paramType.type.includes('UINT')) {
+        if (input < 0) {
+          return 'This parameter must be a positive Integer'
+        }
+      }
+      if (this.edited_param?.paramType.type.includes('INT')) {
+        if (!Number.isInteger(input)) {
+          return 'This parameter must be an Integer'
+        }
+      }
+      return true
+    },
```

Comment:
Should the valid range for the specified type be checked here too (e.g. 0-255 for UINT8), or is that a user problem / expected to get picked up elsewhere?